src/Symfony/Component/Routing/Loader/ControllerAwareXmlFileLoader.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Routing\Loader;
+
+use Symfony\Component\Config\FileLocatorInterface;
+
+/**
+ * Adds support for a "controller" attribute.
+ *
+ * @author Oleg Voronkovich <oleg-voronkovich@yandex.ru>
+ */
+class ControllerAwareXmlFileLoader extends XmlFileLoader
```

Comment:
@stof, The routing component doesn’t know about "controllers", "views", "models" and all things like that. 

> The Routing component maps an HTTP request to a set of configuration variables.

The `controller` keyword is valueless when you use the routing component as a standalone library. So I thinks it's better to extend the existing loaders than change them.